crates/common_utils/src/custom_serde.rs

```diff
@@ -266,6 +266,15 @@ pub mod iso8601custom {
     }
 }
 
+/// serde Serialize using display triat implementation
+pub fn display_serialize<T, S>(value: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    T: std::fmt::Display,
+    S: serde::ser::Serializer,
+{
+    serializer.serialize_str(&format!("{}", value))
+}
```

Comment:
Nit: Spelling: `triat` -> `trait`

Regardless, if the need today is only within the `router` crate, we can consider using [`serde_with::DisplayFromStr`](https://docs.rs/serde_with/3.8.1/serde_with/struct.DisplayFromStr.html) instead of writing an implementation ourselves, and since we pull the dependency already. The example they have includes `mime::Mime` itself!